x11-misc/polybar/polybar-3.1.0.ebuild

```diff
@@ -0,0 +1,64 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+PYTHON_COMPAT=( python2_7 )
+
+inherit cmake-utils python-single-r1
+
+XPP_VERSION="1.4.0"
+I3IPCPP_VERSION="0.7.1"
+
+DESCRIPTION="A fast and easy-to-use tool for creating status bars"
+HOMEPAGE="https://github.com/jaagr/polybar"
+SRC_URI="https://github.com/jaagr/polybar/archive/${PV}.tar.gz -> $P.tar.gz
+	https://github.com/jaagr/xpp/archive/${XPP_VERSION}.tar.gz
+	https://github.com/jaagr/i3ipcpp/archive/v${I3IPCPP_VERSION}.tar.gz
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+IUSE="alsa curl i3wm ipc mpd network"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}
+	dev-lang/python:2.7
+	x11-base/xcb-proto
+	x11-libs/xcb-util-wm
+	x11-libs/xcb-util-image
+	x11-libs/libxcb
+	alsa? ( media-libs/alsa-lib )
+	i3wm? ( dev-libs/jsoncpp )
+	mpd? ( media-libs/libmpdclient )
+	curl? ( net-misc/curl )
+	network? ( net-wireless/wireless-tools )
+"
+DEPEND="${RDEPEND}"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-3.10-fix-generators.patch
+)
+
+src_prepare() {
+	rmdir "${S}"/lib/xpp || die
+	mv "${WORKDIR}"/xpp-$XPP_VERSION "${S}"/lib/xpp || die
+
+	rmdir "${S}"/lib/i3ipcpp || die
+	mv "${WORKDIR}"/i3ipcpp-$I3IPCPP_VERSION "${S}"/lib/i3ipcpp || die
+
+	default
```

Comment:
`default` goes first

Reply:
Patch can't be applied in that case.
It's for a support library and I can't figure out how to apply it in ${WORKDIR} and then move it to ${S}, this way I move it first and then apply it.

Hints how to specify workdir location in the patch file would be appreciated.